WalletWasabi.Gui/Controls/QrCode.cs

```diff
@@ -1,56 +1,221 @@
 using Avalonia;
 using Avalonia.Controls;
 using Avalonia.Media;
+using Avalonia.Media.Imaging;
 using Avalonia.Metadata;
+using Avalonia.Platform;
+using AvalonStudio.Extensibility.Theme;
+using ReactiveUI;
 using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reactive;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
 
 namespace WalletWasabi.Gui.Controls
 {
 	public class QrCode : Control
 	{
+		private static readonly Geometry SaveIcon = Geometry.Parse(
+				"M13,9V3.5L18.5,9M6,2C4.89,2 4,2.89 4,4V20A2,2 0 0,0 6,22H18A2,2 0 0,0 20,20V8L14,2H6Z");
+
+		private ReactiveCommand<Unit, Unit> SaveCommand { get; }
+
+		private const int MatrixPadding = 3;
+
+		private Size CoercedSize = new Size();
+
+		private static DrawingPresenter GetSavePresenter()
+		{
+			return new DrawingPresenter
+			{
+				Drawing = new GeometryDrawing
+				{
+					Brush = Brush.Parse("#22B14C"),
+					Geometry = SaveIcon
+				},
+				Width = 16,
+				Height = 16
+			};
+		}
+
 		static QrCode()
 		{
 			AffectsMeasure<QrCode>(MatrixProperty);
 		}
 
+		public QrCode()
+		{
+			SaveCommand = ReactiveCommand.CreateFromTask(async () =>
+			{
+				await SaveAsync();
+			});
+
+			SaveCommand.ThrownExceptions.Subscribe(ex => Logging.Logger.LogWarning(ex));
+		}
+
+		private async Task SaveAsync()
+		{
+			var source = Matrix;
+
+			if (source is null) return;
+
+			var pixelBounds = PixelSize.FromSize(CoercedSize, 1);
+
+			var sfd = new SaveFileDialog();
+
+			sfd.InitialFileName = $"{Address}.png";
+			sfd.InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.MyPictures);
+			sfd.Filters.Add(new FileDialogFilter() { Name = "PNG Files", Extensions = { "png" } });
+
+			var fileFullName = await sfd.ShowAsync(Application.Current.MainWindow,
+												   fallBack: true);
```

Comment:
Write it in one line: `var fileFullName = await sfd.ShowAsync(Application.Current.MainWindow, fallBack: true);`